### PR TITLE
HLE: Fix integer sign inconcistency accross the codebase

### PR DESCRIPTION
**Pull request by Thog**
This PR is the first step of an overdue code clean up for code from 2018.
Some types were wrongly made signed across all Ryujinx.HLE causing cast all over the place.
This PR address this for memory address and size but also title ids.

As this changes, might bring regressions, this needs intensive testing and careful reviews.

### CODE REVIEW

File: Ryujinx.HLE/HOS/Services/Time/StaticService/ITimeZoneServiceForPsc.cs [hunk 1]
```diff
@@ -188,9 +188,9 @@ public ResultCode GetDeviceLocationNameOperationEventReadableHandle(ServiceCtx c
         // ToCalendarTime(nn::time::PosixTime time, buffer<nn::time::TimeZoneRule, 0x15> rules) -> (nn::time::CalendarTime, nn::time::sf::CalendarAdditionalInfo)
         public ResultCode ToCalendarTime(ServiceCtx context)
         {
-            long posixTime      = context.RequestData.ReadInt64();
-            long bufferPosition = context.Request.SendBuff[0].Position;
-            long bufferSize     = context.Request.SendBuff[0].Size;
+            long posixTime       = context.RequestData.ReadInt64();
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/Time/StaticService/ITimeZoneServiceForPsc.cs:191]: Alignment issue

File: Ryujinx.HLE/HOS/Services/Vi/RootService/IApplicationDisplayService.cs [hunk 2]
```diff
@@ -61,16 +61,16 @@ public ResultCode GetIndirectDisplayTransactionService(ServiceCtx context)
...
             MemoryHelper.FillWithZeros(context.Memory, recBuffPtr, 0x60);
 
             // Add only the default display to buffer
-            context.Memory.Write((ulong)recBuffPtr, Encoding.ASCII.GetBytes("Default"));
-            context.Memory.Write((ulong)recBuffPtr + 0x40, 0x1L);
-            context.Memory.Write((ulong)recBuffPtr + 0x48, 0x1L);
-            context.Memory.Write((ulong)recBuffPtr + 0x50, 1280L);
-            context.Memory.Write((ulong)recBuffPtr + 0x58, 720L);
+            context.Memory.Write(recBuffPtr, Encoding.ASCII.GetBytes("Default"));
+            context.Memory.Write(recBuffPtr + 0x40, 0x1L);
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/Vi/RootService/IApplicationDisplayService.cs:70]: Maybe you could use `UL` for those lines ?

File: Ryujinx.HLE/HOS/Services/Sdb/Pdm/QueryService/QueryPlayStatisticsManager.cs [hunk 3]
```diff
@@ -35,9 +35,9 @@ internal static ResultCode GetPlayStatistics(ServiceCtx context, bool byUserId =
 
             List<ulong> titleIds = new List<ulong>();
 
-            for (int i = 0; i < inputSize / sizeof(ulong); i++)
+            for (ulong i = 0; i < inputSize / sizeof(ulong); i++)
```
[the remark this answers is not in the document]
Comment (Thog) [reply Ryujinx.HLE/HOS/Services/Sdb/Pdm/QueryService/QueryPlayStatisticsManager.cs:38]: avoid a cast of ``inputSize`` here

File: Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/IApplicationFunctions.cs [hunk 4]
```diff
@@ -359,13 +359,13 @@ private ResultCode InitializeApplicationCopyrightFrameBufferImpl(ulong transferM
...
-            ulong frameBufferPos   = context.Request.SendBuff[0].Position;
-            ulong frameBufferSize  = context.Request.SendBuff[0].Size;
-            int  x                 = context.RequestData.ReadInt32();
-            int  y                 = context.RequestData.ReadInt32();
-            int  width             = context.RequestData.ReadInt32();
-            int  height            = context.RequestData.ReadInt32();
-            uint windowOriginMode  = context.RequestData.ReadUInt32();
+            ulong frameBufferPos  = context.Request.SendBuff[0].Position;
+            ulong frameBufferSize = context.Request.SendBuff[0].Size;
+            int  x                = context.RequestData.ReadInt32();
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/Am/AppletOE/ApplicationProxyService/ApplicationProxy/IApplicationFunctions.cs:364]: Wrong alignment

**Review by AcK77 — APPROVED** [review]
Just one nit. Then it LGTM!

File: Ryujinx.HLE/FileSystem/Content/ContentManager.cs [hunk 5]
```diff
@@ -24,8 +24,8 @@ public class ContentManager
 
         private Dictionary<StorageId, LinkedList<LocationEntry>> _locationEntries;
 
-        private Dictionary<string, long>   _sharedFontTitleDictionary;
-        private Dictionary<long, string>   _systemTitlesNameDictionary;
+        private Dictionary<string, ulong>   _sharedFontTitleDictionary;
```
Comment (gdkchan) [inline Ryujinx.HLE/FileSystem/Content/ContentManager.cs:27]: nit: Remove or adjust the alignment here.

**Review by gdkchan — APPROVED** [review]
Looks good other than what I commented 👍

**Pull request merged**